src/r3/atomic-swap/corda/samples/atomic-swap/swap-workflows/src/main/kotlin/com/interop/flows/NotarizationSignaturesCollectorFlow.kt

```diff
@@ -0,0 +1,250 @@
+package com.interop.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import com.r3.corda.evminterop.services.evmInterop
+import com.r3.corda.evminterop.services.swap.DraftTxService
+import com.r3.corda.evminterop.states.swap.LockState
+import net.corda.core.crypto.*
+import net.corda.core.flows.*
+import net.corda.core.identity.AbstractParty
+import net.corda.core.serialization.CordaSerializable
+import net.corda.core.utilities.loggerFor
+import net.corda.core.utilities.unwrap
+import java.math.BigInteger
+import java.security.PublicKey
+
+typealias CollectNotarizationSignaturesFlow = NotarizationSignaturesCollectorFlow.CollectNotarizationSignaturesFlow
+
+object NotarizationSignaturesCollectorFlow {
+
+    @CordaSerializable
+    data class RequestParams(
+        val transactionId: SecureHash,
+        val transactionSignature: TransactionSignature,
+        val notary: PublicKey,
+        val blocking: Boolean
+    )
+
+    @CordaSerializable
+    data class StoreParams(
+        val transactionId: SecureHash,
+        val signature: ByteArray,
+        val notary: PublicKey,
+        val blocking: Boolean
+    ) {
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
+
+            other as StoreParams
+
+            if (transactionId != other.transactionId) return false
+            if (!signature.contentEquals(other.signature)) return false
+            if (notary != other.notary) return false
+            if (blocking != other.blocking) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = transactionId.hashCode()
+            result = 31 * result + signature.contentHashCode()
+            result = 31 * result + notary.hashCode()
+            result = 31 * result + blocking.hashCode()
+            return result
+        }
+    }
+
+    @Suspendable
+    @StartableByRPC
+    @InitiatingFlow
+    class CollectNotarizationSignaturesFlow(
+        val transactionId: SecureHash,
+        val blocking: Boolean
+    ) : FlowLogic<Unit>() {
+
+        companion object {
+            val log = loggerFor<CollectNotarizationSignaturesFlow>()
+        }
+
+        @Suspendable
+        override fun call() {
+            // NOTE -> ME
+            val signedTransaction = serviceHub.validatedTransactions.getTransaction(transactionId)
+                ?: throw IllegalArgumentException("Transaction not found for ID: $transactionId")
+
+            val lockState = signedTransaction.tx.outputs
+                .mapNotNull { it.data as? LockState }
+                .singleOrNull()
+                ?: throw IllegalArgumentException("Transaction $transactionId does not have a lock state")
```

Comment:
Just a note that this exception is slightly wrong as it would be thrown for many lock states too.  Not for this PR but maybe in a further cleanup

Reply:
There can be only one lock state, it is also enforced in the state itself when verifying transaction.

Reply:
Yes, but the exception says there's _no_ lock state.  Which may be wrong if there's multiple states.  So it's misleading.  Not a big deal and probably unlikely to actually be raised.  Just pointing it out